elastic/kibana
# [Fleet] Modify query for Agents bulk actions to exclude unselectable agents

Closes https://github.com/elastic/kibana/issues/152180

## Summary

Modify the query for Agents bulk actions to exclude those agents who are unselectable. 

- In BulkActions component, fetch the agent policies with `is_managed = true`
- Then find the agents that have those policies and are not unenrolled
- Exclude the previous agent ids from the bulk selection query

Doing these two queries first ensures that we have the correct count of agents to exclude so we can subtract it to the total. 

### Testing

- Update an agent policy to be "managed". I did it with the fleet server policy to simulate the behaviour on cloud:
```
PUT kbn:/api/fleet/agent_policies/agent-policy-id
{
  "name": "Managed agent policy",
  "description": "",
  "namespace": "default",
  "monitoring_enabled": [
    "logs",
    "metrics"
  ],
  "is_managed": true
}
```

- Enroll some agents, it's better if they are more than 20 so you can have them on different pages. I used [Horde](https://github.com/elastic/horde) for this step.
- Select all of them, then click on `Select everything on all pages`
- Click on Actions and check that the selected number of agents doesn't include your managed ones
- Select an action and check that the managed actions are not included in the action

https://github.com/elastic/kibana/assets/16084106/cb4cefd6-ca0e-4a72-93c8-19223a491230


### Checklist
